### PR TITLE
[StructuralMechanics] Isotropic Damage CL updates

### PR DESCRIPTION
**Pull request by marandra**
- Renamed from LinearIsotropicDamage to SmallStrainIsotropicDamage
- Fixed moderate-impact bug in hardening law calculation (does not impact values used in test)
- Removed incorrect Has(DAMAGE_VARIABLE) -> true
- Tests updated accordingly

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_constitutive/small_strain_isotropic_damage_plane_strain_2d.cpp [hunk 1]
```diff
@@ -1,100 +1,100 @@
...
+
+//********************************DESTRUCTOR******************************************
+//************************************************************************************
+
+SmallStrainIsotropicDamagePlaneStrain2D::~SmallStrainIsotropicDamagePlaneStrain2D() = default;
+
+//************************************************************************************
+//************************************************************************************
+
+void SmallStrainIsotropicDamagePlaneStrain2D::CalculateElasticMatrix(
```
Comment (loumalouomega) [inline applications/StructuralMechanicsApplication/custom_constitutive/small_strain_isotropic_damage_plane_strain_2d.cpp:46]: I would use the base law instead (deriving from linear elastic). for a different CL

**Review by loumalouomega — APPROVED** [review]
OK. Prevent users of the change please

**Comment by marandra** [comment]
Thanks.
I couldn't find any other use of "LinearIsotropicDamage" in Kratos. I think I'm the only one usong it at the moment

File: applications/StructuralMechanicsApplication/custom_constitutive/small_strain_isotropic_damage_plane_strain_2d.cpp (hunk 1 above)
Comment (marandra) [reply applications/StructuralMechanicsApplication/custom_constitutive/small_strain_isotropic_damage_plane_strain_2d.cpp:46]: Yep. I'll do that in an upcoming PR.

**Pull request merged**